node/src/p2p.rs

```diff
@@ -899,6 +939,55 @@ where
             gossipsub::MessageAcceptance::Ignore
         }
     }
+
+    #[instrument(skip_all)]
+    async fn on_bad_encoding_fraud_sub_message(
+        &mut self,
+        data: &[u8],
+        peer: &PeerId,
+    ) -> gossipsub::MessageAcceptance {
+        let Ok(befp) = BadEncodingFraudProof::decode(data) else {
+            trace!("Malformed or invalid bad encoding fraud proof from {peer}");
+            self.swarm.behaviour_mut().gossipsub.blacklist_peer(peer);
+            return gossipsub::MessageAcceptance::Reject;
+        };
+
+        let height = befp.height().value();
+        let current_height =
+            if let Some(network_height) = network_head_height(&self.header_sub_watcher) {
+                network_height.value()
+            } else if let Ok(local_head) = self.store.get_head().await {
+                local_head.height().value()
+            } else {
+                // we aren't tracking the network and have uninitialized store
+                return gossipsub::MessageAcceptance::Ignore;
+            };
+
+        if height > current_height + FRAUD_PROOF_HEAD_HEIGHT_THRESHOLD {
+            // does this threshold make any sense if we're gonna ignore it anyway
+            // since we won't have the header
+            return gossipsub::MessageAcceptance::Ignore;
+        }
```

Comment:
So how a new node will know that BFEP was published while it was syncing?

Reply:
when we have the fraud-sync, we'll constantly be asking nodes for the fraud proofs, that way we'll catch up with them after we sync enough to be able to validate

Reply:
If my reading is correct, go implementation [only requests proofs from 5 peers ](https://github.com/celestiaorg/go-fraud/blob/main/fraudserv/sync.go#L27) and calls it a day. 
If we want to mirror that it seems like we'd need to fraud-sync after being reasonably cought up with the network, otherwise we'd be dropping proofs on the floor again.

Reply:
hmm I thought it's 5 peers at once but it's a continuously running process :detective: we'll rethink that when we will be writing the sync tho, maybe we will differ in behavior there

Reply:
btw it seems like in some way, those nodes re-announce the proofs when they get them, so maybe we will get those proofs again on the fraud-sub anyway